src/hdskins/java/com/voxelmodpack/hdskins/skins/SkinServer.java

```diff
@@ -28,13 +28,11 @@
             "http://skins.voxelmodpack.com",
             "http://skinmanager.voxelmodpack.com"));
 
-    MinecraftTexturesPayload loadProfileData(GameProfile profile) throws IOException;
+    Map<MinecraftProfileTexture.Type, MinecraftProfileTexture> getProfileTextures(GameProfile profile);
```

Comment:
Don't change this.

Reply:
Why not? It simplifies the rest of those code and the only thing that benefits from there being a separate method is LegacySkinServer, which will already override this function to use its gateway.

Reply:
Legacy has to do several HEAD requests (it does GET, but it can get away with just HEAD) to check for a 404. the method makes that easier. Everything else can do that with a single request (I assume for Bethlehem).

Reply:
That will all happen inside getProfileTextures(), then, which happens either in PreviewTextureManager (which uses a future) or in your cache builder which I can only assume is asynchronous anyway.

The outside world doesn't really need anything more than just a map of textures. (all previous code went through extra hoops to convert it to texture maps which is now done internally by the skin server)